### PR TITLE
revert remove ClusterSet ClusterSetBinding API version v1beta1

### PR DESCRIPTION
**Pull request by haoqing0110**
## Summary
revert the changes in https://github.com/open-cluster-management-io/ocm/pull/257
- add ClusterSet ClusterSetBinding API version v1beta1 back
- add storageversionmigration back
- add conversionwebhook back

## Testing result

StorageVersionMigration append version
```
$ oc get StorageVersionMigration
NAME                                                                   AGE
flowcontrol-flowschema-storage-version-migration                       22h
flowcontrol-prioritylevel-storage-version-migration                    22h
managedclustersetbindings.v1beta1.cluster.open-cluster-management.io   94m
managedclustersets.v1beta1.cluster.open-cluster-management.io          94m
```

crd managedclustersets & managedclustersetbindings
```
status:
  acceptedNames:
    kind: ManagedClusterSet
    listKind: ManagedClusterSetList
    plural: managedclustersets
    shortNames:
    - mclset
    - mclsets
    message: the initial names have been accepted
    reason: InitialNamesAccepted
    status: "True"
    type: Established
  storedVersions:
  - v1beta2

status:
  acceptedNames:
    kind: ManagedClusterSetBinding
    listKind: ManagedClusterSetBindingList
    plural: managedclustersetbindings
    shortNames:
    - mclsetbinding
    - mclsetbindings
    singular: managedclustersetbinding
  conditions:
  - lastTransitionTime: "2023-11-24T06:38:06Z"
    message: no conflicts found
    reason: NoConflicts
    status: "True"
    type: NamesAccepted
  - lastTransitionTime: "2023-11-24T06:38:06Z"
    message: the initial names have been accepted
    reason: InitialNamesAccepted
    status: "True"
    type: Established
  storedVersions:
  - v1beta2
```

create v1beta1 & v1beta2 managedclustersets & managedclustersetbindings, check the result
```
$ cat v1beta1-default.yaml
apiVersion: cluster.open-cluster-management.io/v1beta1
kind: ManagedClusterSet
metadata:
  name: default-v1
spec:
  clusterSelector:
    selectorType: LegacyClusterSetLabel
$ oc create -f v1beta1-default.yaml
W1124 08:41:05.770750  124350 warnings.go:67] cluster.open-cluster-management.io/v1beta1 ManagedClusterSet is deprecated; use cluster.open-cluster-management.io/v1beta2 ManagedClusterSet
managedclusterset.cluster.open-cluster-management.io/default-v1 created
$ cat v1beta2-default.yaml
apiVersion: cluster.open-cluster-management.io/v1beta2
kind: ManagedClusterSet
metadata:
  name: default-v2
spec:
  clusterSelector:
    selectorType: ExclusiveClusterSetLabel
$ oc create -f v1beta2-default.yaml
managedclusterset.cluster.open-cluster-management.io/default-v2 created
$ cat v1beta1-default-binding.yaml
apiVersion: cluster.open-cluster-management.io/v1beta1
kind: ManagedClusterSetBinding
metadata:
  name: default-v1
  namespace: default
spec:
  clusterSet: default-v1
$ oc create -f v1beta1-default-binding.yaml
W1124 08:41:06.235296  124408 warnings.go:67] cluster.open-cluster-management.io/v1beta1 ManagedClusterSetBinding is deprecated; use cluster.open-cluster-management.io/v1beta2 ManagedClusterSetBinding
managedclustersetbinding.cluster.open-cluster-management.io/default-v1 created
$ cat v1beta2-default-binding.yaml
apiVersion: cluster.open-cluster-management.io/v1beta2
kind: ManagedClusterSetBinding
metadata:
  name: default-v2
  namespace: default
spec:
  clusterSet: default-v2
$ oc create -f v1beta2-default-binding.yaml
managedclustersetbinding.cluster.open-cluster-management.io/default-v2 created
```

check kubectl output
```
$ kubectl get mclset
NAME         EMPTY   AGE
default      False   122m
default-v1   True    1s
default-v2   True    1s
global       False   122m
$ kubectl get mclset default-v1 -oyaml | kubectl neat
apiVersion: cluster.open-cluster-management.io/v1beta2
kind: ManagedClusterSet
metadata:
  annotations:
    cluster.open-cluster-management.io/submariner-broker-ns: default-v1-broker
  name: default-v1
spec:
  clusterSelector:
    selectorType: ExclusiveClusterSetLabel
$ kubectl get mclset default-v2 -oyaml | kubectl neat
apiVersion: cluster.open-cluster-management.io/v1beta2
kind: ManagedClusterSet
metadata:
  annotations:
    cluster.open-cluster-management.io/submariner-broker-ns: default-v2-broker
  name: default-v2
spec:
  clusterSelector:
    selectorType: ExclusiveClusterSetLabel
$ kubectl get mclsetbinding default-v1 -oyaml | kubectl neat
apiVersion: cluster.open-cluster-management.io/v1beta2
kind: ManagedClusterSetBinding
metadata:
  name: default-v1
  namespace: default
spec:
  clusterSet: default-v1
$ kubectl get mclsetbinding default-v2 -oyaml | kubectl neat
apiVersion: cluster.open-cluster-management.io/v1beta2
kind: ManagedClusterSetBinding
metadata:
  name: default-v2
  namespace: default
spec:
  clusterSet: default-v2
```

check v1beta1 & v1beta2 webhook
```
$ oc get managedclusterset.v1beta1.cluster.open-cluster-management.io default-v1 -oyaml | kubectl neat
W1124 07:08:50.610067  107634 warnings.go:67] cluster.open-cluster-management.io/v1beta1 ManagedClusterSet is deprecated; use cluster.open-cluster-management.io/v1beta2 ManagedClusterSet
apiVersion: cluster.open-cluster-management.io/v1beta1
kind: ManagedClusterSet
metadata:
  annotations:
    cluster.open-cluster-management.io/submariner-broker-ns: default-v1-broker
  name: default-v1
spec:
  clusterSelector:
    selectorType: LegacyClusterSetLabel

$  oc get managedclusterset.v1beta1.cluster.open-cluster-management.io default-v2 -oyaml | kubectl neat
W1124 07:09:27.401943  107739 warnings.go:67] cluster.open-cluster-management.io/v1beta1 ManagedClusterSet is deprecated; use cluster.open-cluster-management.io/v1beta2 ManagedClusterSet
apiVersion: cluster.open-cluster-management.io/v1beta1
kind: ManagedClusterSet
metadata:
  annotations:
    cluster.open-cluster-management.io/submariner-broker-ns: default-v2-broker
  name: default-v2
spec:
  clusterSelector:
    selectorType: LegacyClusterSetLabel

$ oc get managedclusterset.v1beta2.cluster.open-cluster-management.io default-v1 -oyaml | kubectl neat
apiVersion: cluster.open-cluster-management.io/v1beta2
kind: ManagedClusterSet
metadata:
  annotations:
    cluster.open-cluster-management.io/submariner-broker-ns: default-v1-broker
  name: default-v1
spec:
  clusterSelector:
    selectorType: ExclusiveClusterSetLabel

$ oc get managedclusterset.v1beta2.cluster.open-cluster-management.io default-v2 -oyaml | kubectl neat
apiVersion: cluster.open-cluster-management.io/v1beta2
kind: ManagedClusterSet
metadata:
  annotations:
    cluster.open-cluster-management.io/submariner-broker-ns: default-v2-broker
  name: default-v2
spec:
  clusterSelector:
    selectorType: ExclusiveClusterSetLabel
```

check etcd
```
[root@ip-10-0-135-24 /]# ETCDCTL_API=3 etcdctl --endpoints=${etcd_server_address} --cacert=/etc/kubernetes/static-pod-resources/configmaps/etcd-serving-ca/ca-bundle.crt --key=/etc/kubernetes/static-pod-resources/secrets/etcd-client/tls.key --cert=/etc/kubernetes/static-pod-resources/secrets/etcd-client/tls.crt get /kubernetes.io/cluster.open-cluster-management.io/managedclustersets/default-v1 --prefix --print-value-only | jq .
{
  "apiVersion": "cluster.open-cluster-management.io/v1beta2",
  "kind": "ManagedClusterSet",
  "metadata": {
    "annotations": {
      "cluster.open-cluster-management.io/submariner-broker-ns": "default-v1-broker"
    },
…
    "name": "default-v1",
    "uid": "5b6071c0-7e39-46e4-9c45-672f7f385ea8"
  },
  "spec": {
    "clusterSelector": {
      "selectorType": "ExclusiveClusterSetLabel"
    }
  },
  "status": {
    "conditions": [
      {
        "lastTransitionTime": "2023-11-24T06:39:01Z",
        "message": "No ManagedCluster selected",
        "reason": "NoClusterMatched",
        "status": "True",
        "type": "ClusterSetEmpty"
      }
    ]
  }

[root@ip-10-0-135-24 /]# ETCDCTL_API=3 etcdctl --endpoints=${etcd_server_address} --cacert=/etc/kubernetes/static-pod-resources/configmaps/etcd-serving-ca/ca-bundle.crt --key=/etc/kubernetes/static-pod-resources/secrets/etcd-client/tls.key --cert=/etc/kubernetes/static-pod-resources/secrets/etcd-client/tls.crt get /kubernetes.io/cluster.open-cluster-management.io/managedclustersetbindings/default/default-
v1 --prefix --print-value-only | jq .
{
  "apiVersion": "cluster.open-cluster-management.io/v1beta2",
  "kind": "ManagedClusterSetBinding",
  "metadata": {
    "creationTimestamp": "2023-11-24T08:41:06Z",
    "generation": 1,
    "managedFields": [
      {
        "apiVersion": "cluster.open-cluster-management.io/v1beta1",
        "fieldsType": "FieldsV1",
        "fieldsV1": {
          "f:spec": {
            ".": {},
            "f:clusterSet": {}
          }
        },
        "manager": "kubectl-create",
        "operation": "Update",
        "time": "2023-11-24T08:41:06Z"
      },
      {
        "apiVersion": "cluster.open-cluster-management.io/v1beta2",
        "fieldsType": "FieldsV1",
        "fieldsV1": {
          "f:status": {
            ".": {},
            "f:conditions": {}
          }
        },
        "manager": "registration",
        "operation": "Update",
        "subresource": "status",
        "time": "2023-11-24T08:41:06Z"
      }
    ],
    "name": "default-v1",
    "namespace": "default",
    "uid": "71135620-8af5-4e1c-aec9-f285e103a4e6"
  },
  "spec": {
    "clusterSet": "default-v1"
  },
  "status": {
    "conditions": [
      {
        "lastTransitionTime": "2023-11-24T08:41:06Z",
        "message": "",
        "reason": "ClusterSetBound",
        "status": "True",
        "type": "Bound"
      }
    ]
  }
}
```

### CODE REVIEW

File: manifests/cluster-manager/cluster-manager-managedclustersetbindings-migration.yaml
Comment (elgnay) [inline manifests/cluster-manager/cluster-manager-managedclustersetbindings-migration.yaml:1]: We need to add version info into the name of the migration CR, like managedclustersetbindings.v1beta1.cluster.open-cluster-management.io

File: manifests/cluster-manager/cluster-manager-managedclustersets-migration.yaml
Comment (elgnay) [inline manifests/cluster-manager/cluster-manager-managedclustersets-migration.yaml:1]: We need to add version info into the migration CR, like managedclustersets.v1beta1.cluster.open-cluster-management.io

**Comment by elgnay** [comment]
/override Red Hat Trusted App Test / mce-24-bn6wj / tektok-runner-enterprise-contract-mce-2.4

**Comment by elgnay** [comment]
/override "Red Hat Trusted App Test / mce-24-bn6wj / tektok-runner-enterprise-contract-mce-2.4"

**Comment by elgnay** [comment]
/override "Red Hat Trusted App Test / mce-24-bn6wj / tektok-runner-enterprise-contract-mce-2.4"

File: manifests/cluster-manager/cluster-manager-managedclustersetbindings-migration.yaml
Comment (haoqing0110) [reply manifests/cluster-manager/cluster-manager-managedclustersetbindings-migration.yaml:1]: fixed

File: manifests/cluster-manager/cluster-manager-managedclustersets-migration.yaml
Comment (haoqing0110) [reply manifests/cluster-manager/cluster-manager-managedclustersets-migration.yaml:1]: fixed

**Comment by elgnay** [comment]
/override Red Hat Trusted App Test / mce-24-kf2pg / tektok-runner-enterprise-contract-mce-2.4

**Comment by elgnay** [comment]
/override "Red Hat Trusted App Test / mce-24-kf2pg / tektok-runner-enterprise-contract-mce-2.4"

**Pull request merged**